www/src/views/user.ts

```diff
@@ -3,62 +3,118 @@ import base from './base';
 import Network from '../services/network';
 import Model from '../models/statistics';
 import Chart from 'chart.js';
+import {UserModel} from '../models/user';
 
 class UserProfile implements ξ.ClassComponent {
-  view(vnode: ξ.CVnode) {
-    const username = vnode.attrs.username;
-    const fullname = vnode.attrs.fullname;
-    const email = vnode.attrs.email;
+    isFriend: boolean = false;
 
-    return ξ(base,
-        ξ('.ui.text.container', {
-          style: 'margin-top: 1em;',
-        },
-        ξ('h1.ui.header',
-            ξ('img.ui.circular.image', {
-              src: `/user/${username}/avatar`,
-            }),
-            ξ('.content', `${username}\'S PROFILE`.toUpperCase(),
-                ξ('.sub.header', `You are viewing ${fullname}'s profile.`)),
-        ),
-        ξ('.ui.divider')),
-        ξ('.ui.main.text.container.two.column.stackable.grid',
-            ξ('.ui.column',
-                ξ('img.ui.image.rounded.medium#picture',
-                    {src: '/user/' + username + '/avatar'},
-                    'User profile picture.'),
-                ξ('h2',
-                    fullname),
-                ξ('p',
-                    ξ('i.user.icon'),
-                    username),
-                ξ('p',
-                    ξ('i.mail.icon'),
-                    email),
-                ξ('.ui.placeholder',
-                    ξ('ui.image')),
+    hasFriend(username: string) {
+      for (const friend of UserModel.friends) {
+        if (friend.friend == username) {
+          this.isFriend = true;
+          break;
+        } else {
+          this.isFriend = false;
+        }
+      }
+    };
 
-            ),
-            ξ('.ui.column[min-height = 10em]',
-                ξ('canvas#chart-area', {
-                  oncreate: ({dom}) => {
-                    const ctx = (document.getElementById(
-                        'chart-area') as HTMLCanvasElement)
-                        .getContext('2d');
-                    new Chart(ctx, {
-                      type: 'doughnut',
-                      data: Model.dataDonut,
-                      options: Model.optionsDonut,
-                    });
-                  }}),
-            ),
-        ),
-    );
-  }
+    async friendCheck(username: string) {
+      return Network.request<Array<Friend>>('GET', '/user/' +
+            UserModel.getUsername() + '/friend')
+          .then((result) => {
+            UserModel.emptyFriends();
```

Comment:
I think this is unnecessary, because the next line overwrites the array anyhow?

Reply:
Okay, this is because i was uncertain of the assignment of javascript arrays. I wanted to be sure that the list in the frontend was exactly the same as the backend. And not accidentally do [friend x, friend y] -> get_friends request -> [friend x, friend y, friend x, friend y].

Will remove it ! =)

Reply:
Hehe, yes I'll overwrite it :)